src/textual/widget.py

```diff
@@ -2459,6 +2453,7 @@ async def _scroll_widget_to_center_of_self(
                 duration=duration,
                 easing=easing,
                 force=force,
+                top=origin_visible and widget.region.height > container.region.height,
```

Comment:
The reason I didn't do this approach was that I didn't think it worked for horizontal scrolling.